graphics/managed_surface.cpp

```diff
@@ -386,18 +387,15 @@ void ManagedSurface::blitFromInner(const Surface &src, const Common::Rect &srcRe
 					bDest = bSrc;
 				} else {
 					// Partially transparent, so calculate new pixel colors
-					if (destFormat.bytesPerPixel == 2) {
-						uint32 destColor = *reinterpret_cast<uint16 *>(destVal);
-						destFormat.colorToARGB(destColor, aDest, rDest, gDest, bDest);
-					} else if (format.bytesPerPixel == 4) {
-						uint32 destColor = *reinterpret_cast<uint32 *>(destVal);
-						destFormat.colorToARGB(destColor, aDest, rDest, gDest, bDest);
-					} else {
-						aDest = 0xFF;
-						rDest = destVal[0];
-						gDest = destVal[1];
-						bDest = destVal[2];
-					}
+					uint32 destColor;
+					if (destFormat.bytesPerPixel == 2)
+						destColor = *reinterpret_cast<uint16 *>(destVal);
+					else if (format.bytesPerPixel == 4)
```

Comment:
Oh, while we're here - this reference to format probably should be the local destFormat like the previous check

Reply:
copy-paste error again, fixed in amended commit